imports/plugins/core/versions/server/migrations/72_add_fulfillmentType_to_shipping_methods.js

```diff
@@ -0,0 +1,44 @@
+import { Migrations } from "meteor/percolate:migrations";
+import rawCollections from "/imports/collections/rawCollections";
+
+Migrations.add({
+  version: 72,
+  async up() {
+    const { Cart, Shipping } = rawCollections;
+
+    await Shipping.updateMany({
+      "provider.name": "flatRates"
+    }, {
+      $set: {
+        "methods.$[].fulfillmentTypes": ["shipping"]
+      }
+    });
+
+    await Cart.updateMany({}, {
+      $set: {
+        "shipping.$[].shipmentMethod.fulfillmentTypes": ["shipping"],
+        "shipping.$[groupWithShipmentQuotes].shipmentQuotes.$[].method.fulfillmentTypes": ["shipping"]
+      }
+    }, {
+      arrayFilters: [{ "groupWithShipmentQuotes.shipmentQuotes": { $exists: true } }]
+    });
+  },
+  async down() {
+    const { Cart, Shipping } = rawCollections;
+
+    await Shipping.updateMany({
+      "provider.name": "flatRates"
+    }, {
+      $unset: {
+        "methods.$[].fulfillmentTypes": ""
+      }
+    });
+
+    await Cart.updateMany({}, {
+      $unset: {
+        "shipping.$[].shipmentMethod.fulfillmentTypes": "",
+        "shipping.$[].shipmentQuotes.$[].method.fulfillmentTypes": ""
+      }
+    });
```

Comment:
This down update will also need the arrayFilters fix.

```js
    await Cart.updateMany({}, {
      $unset: {
        "shipping.$[].shipmentMethod.fulfillmentTypes": "",
        "shipping.$[groupWithShipmentQuotes].shipmentQuotes.$[].method.fulfillmentTypes": ""
      }
    }, {
      arrayFilters: [{ "groupWithShipmentQuotes.shipmentQuotes": { $exists: true } }]
    });
```

Reply:
added, thanks